bok_choy/page_object.py

```diff
@@ -330,6 +345,25 @@ def _verify_page(self):
             )
             raise WrongPageError(msg)
 
+    def _verify_xss_exposure(self):
+        """
+        Verify that there are no obvious XSS exposures on the page (based on test authors
+        including XSS_INJECTION in content rendered on the page).
+
+        If an xss issue is found, raise a 'XSSExposureError'.
+        """
+        # Use innerHTML to get dynamically injected HTML as well as server-side HTML.
+        html_source = self.browser.execute_script(
+            "return document.documentElement.innerHTML.toLowerCase()"
+        )
+
+        # Check taken from https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#XSS_Locator.
+        all_hits = html_source.count('<xss')
+        if all_hits > 0:
+            value_hits = len(re.findall(r'value="\'\';!--.*<xss.*{\(\)}"', html_source))
+            if all_hits > value_hits:
+                raise XSSExposureError("{} XSS issue(s) found on page".format(all_hits - value_hits))
```

Comment:
Couple ideas:
* We might give #85 some love. That way the person could see the entire source of the rendered page. Not sure though if it's not helpful unless we use the document.documentElement.innerHTML trick from above.
* Something like below would at least narrow down to where they were on the page. This assumes (I’m not sure if it’s a valid assumption) that the vulnerability will be in the HTML after another occurrence of “<”. I think this might hold true because in HTML you’re at least past the body tag and in JS past a script tag.
```
        all_hits_count = html_source.count('<xss')
        if all_hits_count > 0:
            safe_hits = re.findall(r'value="\'\';!--.*<xss.*{\(\)}"', html_source)
            safe_hits_count = len(safe_hits)
            if all_hits_count > safe_hits_count:
                potential_hits = re.findall('<[^<]+<xss', html_source)
                raise XSSExposureError(
                    "{} XSS issue(s) found on page. Potential places are {}".format(all_hits_count - safe_hits_count, potential_hits)
                )
```